controlplane/kubeadm/internal/control_plane.go

```diff
@@ -238,6 +238,13 @@ func (c *ControlPlane) IsEtcdManaged() bool {
 	return c.KCP.Spec.KubeadmConfigSpec.ClusterConfiguration == nil || c.KCP.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd.External == nil
 }
 
+// UnhealthyMachinesWithNonMHCUnhealthyCondition returns all unhealthy control plane machines that
+// does not have a Kubernetes node or have any control plane component condition set to False.
+// It is different from the UnhealthyMachines func which checks MachineHealthCheck conditions.
+func (c *ControlPlane) UnhealthyMachinesWithNonMHCUnhealthyCondition(machines collections.Machines) collections.Machines {
```

Comment:
I'm confused by the name of this method and how it maps to the implementation
The name mentions unhealthy machines but not marked by MHC as unhealthy
However, the implementation is just checking for the health of the CP components pods.

Could this maybe be:
```suggestion
func (c *ControlPlane) MachinesWithUnhealthyPods(machines collections.Machines) collections.Machines {
```
Just a suggestion, there might be better names

Reply:
Thanks. We could have a better func name. 
btw would machines with unhealthy control plane component pods lead to MachineHealthCheckSucceededCondition set to false?

Reply:
Good question, I'm not completely sure. I don't remember all the MHC code from the top of my head, I would just test it empirically.

Reply:
> Thanks. We could have a better func name.
> btw would machines with unhealthy control plane component pods lead to MachineHealthCheckSucceededCondition set to false?

I don't think so. Here is the code: https://github.com/kubernetes-sigs/cluster-api/blob/1c760a5953cc9af2a78a51b943eb19dc9ead0a33/internal/controllers/machinehealthcheck/machinehealthcheck_targets.go#L92

I think the Machine would only be remediated if:
* the corresponding node doesn't show up (e.g. because of the unhealthy control plane component somehow)
* the unhealthy control plane component pod somehow leads to a specific Node condition and the MHC is configured to react on that